packages/apps/docs/src/components/InfiniteScroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,53 @@
+import { Button, Heading, Stack, Text } from '@kadena/react-ui';
+
+import { Loading } from '@/components';
+import React, { FC, useEffect, useRef } from 'react';
+
+interface IProps {
+  handleLoad: () => void;
+  handleRetry: () => void;
+  isLoading: boolean;
+  error?: string;
+  isDone?: boolean;
+}
+
+export const InfiniteScroll: FC<IProps> = ({
+  handleLoad,
+  handleRetry,
+  isLoading,
+  error,
+  isDone,
+}) => {
+  const loadRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
```

Comment:
can look into returning a cleanup function in the `useEffect` to clean up the `IntersectionObserver`

Reply:
done